plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -1114,6 +1116,12 @@ public String quoted(RemoteTableName remoteTableName)
         return emptyMap();
     }
 
+    @Override
+    public String canonicalize(ConnectorSession session, String value, boolean delimited)
+    {
+        return value.toLowerCase(ENGLISH);
```

Comment:
I think you should use `connection.getMetaData().storesUpperCaseIdentifiers();` and so you might need https://github.com/trinodb/trino/pull/7721

Reply:
Thats the final implementation once we fix it for all the JDBC based connector.  For now most of the JDBC connector works in legacy way